internal/resolver/dns/dns_resolver.go

```diff
@@ -201,18 +207,49 @@ func (d *dnsResolver) Close() {
 
 func (d *dnsResolver) watcher() {
 	defer d.wg.Done()
+	backoffIndex := 1
+	nextBackoffRate := backoff.DefaultExponential.Backoff(backoffIndex)
 	for {
 		select {
 		case <-d.ctx.Done():
+			d.pollTimer.Stop()
 			return
 		case <-d.rn:
+			// Reset polling timer if active, as ResolveNow was called from ClientConn.
+			if !d.pollTimer.Stop() {
+				select {
+				case <-d.pollTimer.C:
+				default:
+				}
```

Comment:
I don't think you want this `select`.  As mentioned below, it might be  possible `Stop` returns false but hasn't _yet_ pushed to the channel (but _will_).  Just do a straight blocking pull from the timer channel in this case.

Reply:
This is my main issue. How would a blocking pull from a timer channel not inherently cause this to sit in this state forever in some cases? If the pollTimer has already been drained and not reset (your suggestion, remember I was creating new memory before), t.Stop() will return false and never send on the channel right?

Reply:
Discussed in chat. This was solved by setting to max duration on success case.